internal/corazawaf/waf.go

```diff
@@ -267,15 +269,25 @@ func NewWAF() *WAF {
 		// Initializing pool for transactions
 		txPool: sync.NewPool(func() interface{} { return new(Transaction) }),
 		// These defaults are unavoidable as they are zero values for the variables
-		RuleEngine:               types.RuleEngineOn,
-		RequestBodyAccess:        false,
-		RequestBodyLimit:         _1gb,
-		RequestBodyInMemoryLimit: _1gb,
-		ResponseBodyAccess:       false,
-		ResponseBodyLimit:        _1gb,
-		AuditLogWriter:           logWriter,
-		Logger:                   logger,
+		RuleEngine:         types.RuleEngineOn,
+		RequestBodyAccess:  false,
+		RequestBodyLimit:   _1gb,
+		ResponseBodyAccess: false,
+		ResponseBodyLimit:  _1gb,
+		AuditLogWriter:     logWriter,
+		Logger:             logger,
+		TmpDir:             os.TempDir(),
 	}
+
+	// Here we link the request body in memory limit with the request body limit,
+	// hence it changes accordingly unless the user sets the memory limit to differ.
+	// This allows the WAF object to be valid despite wether the request body in memory
+	// limit is set or not and only perform the comparisons when the value is set by the
+	// user. For example, the test could require the requestBodyLimit to be 3 and no need
+	// to worry about the requestBodyInMemoryLimit but the WAF sill be invalid and the
+	// validation will fail.
+	waf.requestBodyInMemoryLimit = &(waf.RequestBodyLimit)
```

Comment:
This pointer aliasing is hard to reason about and doesn't seem very idiomatic Go I think. While yes the name `Validate()` doesn't seem like a mutating method, because it's internal code I think it's OK for it to just update the in memory limit if it hasn't been set, it will still need an explanatory comment but I think easier to understand then pointer alias